TaskManager.java

```diff
@@ -0,0 +1,71 @@
+@Component
+public class TaskManager {
+    @PersistenceContext
+    private EntityManager entityManager;
+    private static TaskManager instance;
```

Comment:
Why we need a reference to instance if we already declared this class to be a @Component, Singleton by default